linode/packer-plugin-linode
# Add image replicate support

## 📝 Description

This is to support image replication feature brought by the Image gen2 project.

## ✔️ How to Test

### Automated Testing

```bash
make test
```

### Manual Testing

Checkout this branch and then install the packer plugin by:

```bahs
make dev
```

Create a file named `test.pkr.hcl` with the following content.

```hcl
locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }

source "linode" "example" {
  image             = "linode/debian11"
  image_description = "My Private Image"
  image_label       = "private-image-${local.timestamp}"
  instance_label    = "temporary-linode-${local.timestamp}"
  instance_type     = "g6-nanode-1"
  region            = "us-mia"
  ssh_username      = "root"

  image_regions     = ["us-mia", "in-maa", "us-ord", "us-lax", "us-sea", "jp-osa", "it-mil", "nl-ams", "se-sto"]

  interface {
    purpose = "public"
  }
}

build {
  sources = ["source.linode.example"]
}
```

Finally, run the build command:
```
packer build .
```

Verify the image has been replicated to the selected regions in `image_regions` attribute in the HCL file above.

<img width="464" alt="image" src="https://github.com/user-attachments/assets/18a72b56-83af-43a1-a125-9ef666df34c3" />
